legacy/build-deploy-docker-compose.sh

```diff
@@ -698,6 +698,28 @@ if [[ "$BUILD_TYPE" == "pullrequest"  ||  "$BUILD_TYPE" == "branch" ]]; then
         echo "defined Dockerfile $DOCKERFILE for service $IMAGE_NAME not found"; exit 1;
       fi
 
+      set +x # reduce noise in build logs
+      # Decide whether to use BuildKit for Docker builds - disabled by default.
+      DOCKER_BUILDKIT=0
+
+      if [ ! -z "$LAGOON_PROJECT_VARIABLES" ]; then
+        DOCKER_BUILDKIT=($(echo $LAGOON_PROJECT_VARIABLES | jq -r '.[] | select(.scope == "build") | select(.name == "DOCKER_BUILDKIT") | "\(.value)"'))
+      fi
+      if [ ! -z "$LAGOON_ENVIRONMENT_VARIABLES" ]; then
+        DOCKER_BUILDKIT=($(echo $LAGOON_ENVIRONMENT_VARIABLES | jq -r '.[] | select(.scope == "build") | select(.name == "DOCKER_BUILDKIT") | "\(.value)"'))
+      fi
```

Comment:
[You need to do this sort of pattern](https://github.com/uselagoon/build-deploy-tool/blob/v0.19.4/legacy/build-deploy-docker-compose.sh#L269-L277). Otherwise it will almost always be 0 due to how the two if statements evaluate.

Reply:
I'll relook at it - technically it does work, because the environment not being set doesn't evaluate as truthy, so gets reset to 0 anyway.

Reply:
See these two examples https://gist.github.com/shreddedbacon/f418d84c7e3feff47bdc241308a503ea

Reply:
I just added a third example using the TEMP_X pattern to show you the difference between

Reply:
ugh :facepalm: - of course, if there's no ENV vars set, it gets skipped, hence my pass!